File: backend/src/main/java/io/github/xpakx/ladder/entity/dto/ProjectData.java
package io.github.xpakx.ladder.entity.dto;

import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
public class ProjectData {
    private ProjectDetails project;
    private List<TaskDetails> tasks;
    private List<HabitDetails> habits;
}

File: frontend/src/app/project/dto/project-data.ts
import { HabitDetails } from "src/app/habit/dto/habit-details";
import { ProjectDetails } from "./project-details";
import { TaskDetails } from "src/app/task/dto/task-details";

export interface ProjectData {
    project: ProjectDetails;
    tasks: TaskDetails[];
    habits: HabitDetails[];
    //habitCompletions: HabitCompletionDetails[];
}